runtime/lib/Handlers/setV3/lib/evaluateScript.ts

```diff
@@ -0,0 +1,60 @@
+import { CompiledSetV3NodeDataItem, SetV3NodeDataItemType } from '@voiceflow/dtos';
+import _ from 'lodash';
+
+import Store from '@/runtime/lib/Runtime/Store';
+import { toErrorReasonString } from '@/utils/error/toErrorReasonString';
+import { joinTokens } from '@/utils/string/joinTokens';
+
+import { IVMIsolate } from '../../utils/ivmIsolate';
+import { SetEvaluator } from './setEvaluator';
+import { getSetCode, getSetVariable } from './utils';
+
+async function evaluate(
+  isolate: IVMIsolate,
+  item: CompiledSetV3NodeDataItem,
+  updatePatch: Record<string, unknown>,
+  logToPrototype: (message: string) => void
+) {
+  try {
+    if (item.type !== SetV3NodeDataItemType.SCRIPT) {
+      throw new Error(`received the wrong type of set assignment, received type '${item.type}'`);
+    }
+
+    if (!item.value) {
+      throw new Error(`code is undefined`);
+    }
+
+    if (!item.variable) {
+      throw new Error(`target variable is undefined`);
+    }
+
+    const value = await isolate.executeFunctionOrScript(item.value);
+    updatePatch[item.variable] = value;
```

Comment:
```suggestion
    return { [item.variable]: await isolate.executeFunctionOrScript(item.value) };
```
and shouldn't take `updatePatchItem` as a param. see below comment for reasoning